TelegramBot/TelegramBot/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Text;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks; //TemporaryFunctions
+
+namespace TelegramBot.Repositories
+{
+    public class BaseRepository<TEntity, TKey>
+    {
+        public Uri Root { get; set; }
+        public string Token { get; set; }
+        private readonly HttpClient httpClient = new();
+
+        public BaseRepository(string entityRoot)
+        {
+            Root = new Uri(AppSettings.BaseRoot + entityRoot);
+            Token = AppSettings.TokenApi;
+
+            HttpClientHandler handler = new();
+            handler.CookieContainer = new CookieContainer();
+            handler.CookieContainer.Add(this.Root, new Cookie("token", Token));
+            httpClient = new HttpClient(handler);
+        }
+
+        public async Task<TEntity> Get(TKey key)
+        {
+            HttpResponseMessage httpResponse = await httpClient.GetAsync(Root + "/" + key);
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                return await Deserialize<TEntity>(httpResponse);
+            }
+            throw new Exception("Get(TKey key): - StatusCode - "
+                                + httpResponse.StatusCode
+                                + " Content - "
+                                + await httpResponse.Content.ReadAsStringAsync());
+
+        }
+
+        public async Task<TEntity> GetCurrentUser()
+        {
+            HttpResponseMessage httpResponse = await httpClient.GetAsync(Root);
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                return await Deserialize<TEntity>(httpResponse);
+            }
+            throw new Exception("Get(TKey key): - StatusCode - "
+                                + httpResponse.StatusCode
+                                + " Content - "
+                                + await httpResponse.Content.ReadAsStringAsync());
+        }
+    
+        public async Task<List<TEntity>> Get()
+        {
+            HttpResponseMessage httpResponse = await httpClient.GetAsync(Root);
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                return await Deserialize<List<TEntity>>(httpResponse);
+            }
+            throw new Exception("Get(): - StatusCode - "
+                                + httpResponse.StatusCode
+                                + " Content - "
+                                + await httpResponse.Content.ReadAsStringAsync()); 
+        }
+        public async Task<HttpResponseMessage> Delete(TKey key) 
```

Comment:
Зачем возвращать сообщение об ошибке?
```suggestion
        public async Task Delete(TKey key) 
```
Лучше просто выкидывать исключение
Добавь [кастомное исключение](https://metanit.com/sharp/tutorial/3.17.php), добавь под него отдельную папку и назови ее `Interfaces`, а само исключение - `ErrorResponseExceprion`